### DISCUSSION TITLE
Github Release notes/Changelog fetching is not working

### DISCUSSION THREAD

**Discussion by otbe** [renovatebot/renovate#11129] [Request Help]
**How are you running Renovate?**

- [ ] WhiteSource Renovate hosted app on github.com
- [x] Self hosted

If using the hosted app, please skip to the next section.
Otherwise, if self-hosted, please complete the following:

Please select which platform you are using:

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server
- [ ] Gitea
- [ ] github.com
- [ ] GitHub Enterprise Server
- [x] gitlab.com
- [ ] GitLab self-hosted

Renovate version: renovate/renovate:25.69.0

Since some times our renovate PRs do not include any information about the version. For example updates for spring/boot or testcontainers usually show a list of changes in this version. I recently updated our renovate bot to use the version above in order to fix it but it stayed the same.  We've a github token in place to workaround their API limitations. Its within our renovate.json and looks like this

```json
{
  "extends": ["config:base", "docker:enableMajor"],
  "hostRules": [
    {
      "domainName": "github.com",
      "encrypted": {
        "token": "superSecretToken"
      }
    },
   ]
}
```
I can see log messages like the following

```
DEBUG: getReleaseList 404 (repository=testcontainers/testcontainers-java)
       "url": "https://api.github.com/projects/testcontainers%2ftestcontainers-java/releases?per_page=100"
DEBUG: GitLab API 404 (repository=projectOnGitlabCom)
DEBUG: Error 404 getting changelog md (repository=projectOnGitlabCom)
       "url": "https://api.github.com/projects/testcontainers%2ftestcontainers-java/repository/tree?per_page=100"
```

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [x] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

**Comment by viceice** [comment]
You can't use encryption on self-hosted renovate by default. You need to configure it

https://docs.renovatebot.com/self-hosted-configuration/#privatekey

**Reply by otbe** [reply]
Yes it is configured with a custom private key. We have custom ecr and gitlab container registry credentials setup this way.
But I think I found the solution. I switched from a hostrule for github.com to `GITHUB_COM_TOKEN` via env to provide the token and now it works. 

Thanks
